R/checks.R

```diff
@@ -0,0 +1,60 @@
+#' Check if offspring argument is specified as a character string
+#'
+#' @param offspring_dist Offspring distribution: a character string
+#' corresponding to the R distribution function (e.g., "pois" for Poisson,
+#' where \code{\link{rpois}} is the R function to generate Poisson random
+#' numbers).
+#' @keywords internal
+check_offspring_valid <- function(offspring_dist) {
+  if (!checkmate::test_string(offspring_dist)) {
+    stop(sprintf(
+      "%s %s",
+      "'offspring_dist' must be specified as a character string.",
+      "Did you forget to enclose it in quotes?"
+    ))
+  }
+}
+
+
+#' Check if constructed random number generator for offspring exists
```

Comment:
Isn't this the same as `check_serial_valid` and could it be a single function `check_func_valid` (which we could also use for the `_ll` functions).

Reply:
This has been converted to an issue here #47.